controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -277,6 +282,17 @@ func (r *StorageConsumerReconciler) reconcileCephBlockPool() error {
 	failureDomain = storageClusterList.Items[0].Status.FailureDomain
 
 	capacity := r.storageConsumer.Spec.Capacity.String()
+	deviceSetList := storageClusterList.Items[0].Spec.StorageDeviceSets
+	var deviceSet *ocsv1.StorageDeviceSet = nil
+	for _, ds := range deviceSetList {
+		if ds.Name == StorageDeviceName {
+			deviceSet = &ds
+		}
+	}
+	pgUnitSize := MaxPGUnitSize
+	if deviceSet != nil {
+		pgUnitSize = getUnitSizeOfPgs(deviceSet.Count)
+	}
```

Comment:
This is wrong. If you cannot find the correct device set in provider mode you should return an error and not continue reconciling as this is an erroneous state